src/mame/machine/dec_lk201.cpp

```diff
@@ -549,25 +553,34 @@ void lk201_device::device_timer(emu_timer &timer, device_timer_id id, int param,
 		break;
 
 	case 2:
-		m_speaker->set_output_gain(0, 0);
-		m_speaker->set_state(0);
+		if (m_beep_state == 2)
+		{
+			m_beeper->adjust(attotime::never);
+
+			m_speaker->set_state(0);
+			m_beep_state = 0;  // ready for new beep.
+
+		} else
+			m_beeper->adjust(attotime::from_msec(20));
+		
 		break;
 
 	default:
 		device_serial_interface::device_timer(timer, id, param, ptr);
+	}
 }
-}
-
 
 void lk201_device::rcv_complete()
 {
 	sci_status |= SCSR_RDRF;
 	update_interrupts();
 	receive_register_extract();
 
-	int data = get_received_char();
-	m_kbd_state = data;
-//  printf("\nlk201 got %02x\n", m_kbd_state);
+	m_kbd_state = get_received_char();
+//	printf("\nlk201 got %02x\n", m_kbd_state);
+
+	if(m_kbd_state == 0xfd)
```

Comment:
Please be consistent with spacing.  You have no space between if and the opening parenthesis, while you have a space there in other places in this file.